app/client/components/LinkingState.ts

```diff
@@ -194,13 +200,111 @@ export class LinkingState extends Disposable {
+      //                   [ A ]--------/        [ B ]   --------------/       [ C ]                                   |
+      //                        A.actRowId                B.actRowId                                                   |
+      //
+      // However, if e.g. viewSec B is filtered, the correct rowId might not exist in B, and so it's rowId would be
+      // on a different row, and therefore the cursor linking would set C to a different row from A, even if it existed
+      // in C
+      //
+      // Normally this wouldn't be too bad, but to implement bidirectional linking requires allowing cycles of
+      // cursor-links, in which case this behavior becomes extra-problematic, both in being more unexpected from a UX
+      // perspective and because a section will eventually be linked to itself, which is an unstable loop.
+      //
+      // A better solution is to propagate the linked rowId directly through the chain of linkingStates without passing
+      // through the activeRowIds of the sections, so whether a section is filtered or not doesn't affect propagation.
+      //
+      //                                                B.LS.incCursPos                                                |
+      //                                 -->  [B.LS]   -------------->   [C.LS]                                        |
+      //                                /        |                          |                                          |
+      //                               /         v B.LS.cursorPos           v C.LS.cursorPos                           |
+      //                 [ A ]--------/        [ B ]                      [ C ]                                        |
+      //                      A.actRowId                                                                               |
+      //
+      // If the previous section has a linkingState, we use the previous LS's incomingCursorPos
+      // (i.e. two sections back) instead of looking at our srcSection's activeRowId. This way it doesn't matter how
+      // section B is filtered, since we're getting our cursorPos straight from A (through a computed in B.LS)
+      //
+      // However, each linkingState needs to decide whether to use the cursorPos from the srcSec (i.e. its activeRowId),
+      // or to use the previous linkState's incomingCursorPos. We want to use whichever section the user most recently
+      // interacted with, i.e. whichever cursor update was most recent. For this we use, the cursor version (given in
+      // viewSection.lastCursorEdit). incomingCursorPos is a pair of [rowId, sequenceNum], so each linkingState sets its
+      // incomingCursorPos to whichever is most recent between its srcSection, and the previous LS's incCursPos.
+      //
+      // The end result is that there is that because the lastCursorEdits are guaranteed to be unique, there is always
```

Comment:
```suggestion
      // The end result is that because the lastCursorEdits are guaranteed to be unique, there is always
```
But not sure what this means